### DISCUSSION TITLE
Flutter Web/Provider Running out of Memory

### DISCUSSION THREAD

**Discussion by kornha** [rrousselGit/provider#743] [Q&A]
Hello,

I hope this is appropriate to mention here; I believe I am running into an out of memory error with Provider when combined with Flutter Web and Go Router.

I have setup my User class as a ChangeNotifierProvider, which gets notified from a StreamListener, as shown below.

```
class User extends ChangeNotifier {
...

  User({required this.uid}) {
    Database().getUser(uid).listen(
      (user) async {
        displayName = user?.displayName;
        email = user?.email;
        phoneNumber = user?.phoneNumber;
        photoURL = user?.photoURL;
        did = user?.did;
        interests = user?.interests;
        notifyListeners();
      },
      onError: (e) => print(e),
    );
  }
...
}
```
In my main.dart, I set my providers and router as such, using the Flutter go_router (v2, not 3).
```
return MultiProvider(
  providers: [
    ChangeNotifierProvider<AuthState>.value(value: _authState),
    ChangeNotifierProvider<ThemeModel>(create: (_) => ThemeModel())
  ],
  child: Consumer<AuthState>(
    builder: (context, auth, child) {
      var user =
          auth.authUser == null ? null : User(uid: auth.authUser!.uid);
      return MultiProvider(
        providers: [
          ChangeNotifierProvider<ZUser?>.value(
            value: zuser,
          ),
        ],
        child: MaterialApp.router(...
```
I use Firestore to update the DB, as such:
```
  Future updateUser(String uid, Map<String, Object?> data) async {
    return userCollection.doc(uid).update(data);
  }
```
Called from:
```
ZWideButton(
  text: "Save",
  onPressed: () async {
    setState(() {
       _localEdit = false;
       _loading = true;
    });
    await user.saveInterests(_interests());
    setState(() => _loading = false);
  },
),
```
intermediate method:
```
 Future saveInterests(List<String> interests) async {
    return _db.updateUser(uid, {"interests": interests});
  }
```
This works as expected. However, on Flutter web, if I repeatedly make calls I run out of memory. Furthermore, if I leave open the page for a while I also sometimes run out of memory. I am suspect that this is caused by Provider perhaps interacting with the GoRouter, but I cannot be certain, it is very difficult to ascertain. I thought this may be caused by the location of my Provider within the tree, namely at the top ahead of the Material router, however this bug exists even if I move it down lower. I now suspect there is some issue with the User not being disposed appropriately.

Any thoughts? Is this the right place to look?

**Comment by rrousselGit** [comment]
I doubt that this is related to Provider. 
There's not really a place where memory would keep pilling up.

I'd suggest using the Flutter devtool to investigate the memory usage, to see what's taking up all the memory

**Reply by kornha** [reply]
I appreciate the response. Sadly, I cannot profile memory on flutter web as that functionality does not currently work. I will look at other causes; following up on stackoverflow here for those interested: https://stackoverflow.com/questions/71559847/flutter-web-firestore-running-out-of-memory-provider

**Reply by tsinis** [reply]
Hey! Can you provide a reproducible example or test it on desktop/mobile platforms with dev tools?
